py3status/modules/velib_metropole.py

```diff
@@ -0,0 +1,292 @@
+    {station_name}              station location name, eg 'Enghien - Faubourg Poissonnière'
+    {station_state}             current station state, eg 'Operative'
+    {station_state_code}        current station state code, eg '0'
+    {station_type}              station type, eg 'yes' (?)
+
+"""
+from datetime import datetime
+from re import sub
+
+STRING_MISSING_STATIONS = "No velib stations set"
+
+
+class Py3status:
+    """
+    """
+
+    # available configuration parameters
+    button_next = 5
+    button_previous = 4
+    button_refresh = 2
+    cache_timeout = 60
+    format = "Velib Metropole: {index}/{stations} {format_station}"
+    format_station = ("{station_name}: [\?color=station_state_code {station_state}]"
+                      "[\?soft  ][\?color=greenyellow {nb_bike}/{nb_free_e_dock}]"
+                      "[\?soft  ][\?color=deepskyblue {nb_ebike}/{nb_free_e_dock}]")
+    station_codes = [20043, 11014, 20012, 20014, 10042]
+    thresholds = {"station_state_code": [(0, "good"), (1, "bad")]}
+
+    class Meta:
+        update_config = {
+            "update_placeholder_format": [
+                {
+                    "placeholder_formats": {
+                        "density_level": ":.0f",
+                        "max_bike_overflow": ":.0f",
+                        "nb_bike": ":.0f",
+                        "nb_bike_overflow": ":.0f",
+                        "nb_dock": ":.0f",
+                        "nb_e_bike_overflow": ":.0f",
+                        "nb_e_dock": ":.0f",
+                        "nb_ebike": ":.0f",
+                        "nb_free_dock": ":.0f",
+                        "nb_free_e_dock": ":.0f",
+                        "station_gps_latitude": ":.3f",
+                        "station_gps_longitude": ".3f",
+                        "station_due_date": ".0f",
+                    },
+                    "format_strings": ["format_station"],
+                }
+            ]
+        }
+
+    def post_config_hook(self):
+        if not self.station_codes:
+            raise Exception("%s" % (STRING_MISSING_STATIONS))
+
+        # string to list if necessary
+        if not isinstance(self.station_codes, list):
+            self.station_codes = [self.station_codes]
+
+        self.request_timeout = 10
+
+        # take the whole map for first run
+        # default values take all stations
+        # around Paris
+        # Area is then shrink to only get
+        # specified stations
+        self.gps_top_right_latitude = 49.0
+        self.gps_top_right_longitude = 2.5
+        self.gps_bot_left_latitude = 48.6
+        self.gps_bot_left_longitude = 2.2
+        self.gps_zoom_level = 15
+
+        self.station_states = {"Operative": 0, "Close": 1}
+
+        json = self._get_velib_data()
+        if json:
+            self.number_of_stations, self.stations = self._manipulate(json)
+            self._get_usefull_area(self.stations)
```

Comment:
(Opinion) I'm pretty sure if you used `try/except` + `self.gps` dict to get useful area in code, it would look more clean than reproducing this code again in the `post_config_hook`. Remember `try/except` in `hddtemp`? Do something like that.

Reply:
ok will try to put this at another place

Reply:
done without try/except, hope this is well